python/tests/estimators/test_keras_estimators.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import os
+import shutil
+import tempfile
+
+import PIL.Image
+import numpy as np
+from keras.applications import Xception
+from keras.applications.imagenet_utils import preprocess_input
+
+from sparkdl.estimators.keras_image_file_estimator import KerasImageFileEstimator
+from sparkdl.transformers.keras_image import KerasImageFileTransformer
+from sparkdl.utils.keras_model import *
+
+from ..tests import SparkDLTestCase
+from ..transformers.image_utils import getSampleImagePaths
+
+
+def _load_image_from_uri(local_uri):
+    img = (PIL.Image
+           .open(local_uri)
+           .convert('RGB')
+           .resize((299, 299), PIL.Image.ANTIALIAS))
+    img_arr = np.array(img).astype(np.float32)
+    img_tnsr = preprocess_input(img_arr[np.newaxis, :])
+    return img_tnsr
+
+class KerasEstimatorsTest(SparkDLTestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.model_filename = os.path.join(self.temp_dir, 'model.h5')
+
+        self.estimator = KerasImageFileEstimator(inputCol='imageUri', outputCol='preds',
+                                                 labelCol='label', imageLoader=_load_image_from_uri,
+                                                 optimizer='adam', loss='categorical_crossentropy',
+                                                 kerasFitParams={}, modelFile=self.model_filename)
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir, ignore_errors=True)
+
+    def test_valid_params(self):
+        self.assertTrue(self.estimator._validateParams())
+
+    def test_valid_workflow(self):
+        # Saving a model
+        model = Xception(weights=None)
+        model.save(self.model_filename)
```

Comment:
It's a bit weird to build the estimator and `model_filename` and setUp but not write out the actual model until we're inside the test. If other tests wanted to use this estimator, they'd have to delete this file and replace it.

Why don't we put the estimator creation inside the test for now, we can promote it to setup if we write other tests that need it.